cypress/support/step_definitions/dashboard/there_is_a_list_with_all_states.js

```diff
@@ -66,153 +67,246 @@ function createListForService(service) {
-        data: {
-          pinnedItems: {
-            set: shouldPin,
-          },
-        },
-        where: {
-          email: "smoke@cautionyourblast.com",
-        },
+        reference: "SMOKE"
       },
-    });
-  });
+      include: {
+        items: true
+      }
+    }
+  }).then((list) => {
+    const listId = list.id;
+    const { items } = list;
+    addPins(items);
+  })
+
 });
 
+function setupPublishEvents(options) {
+  const events = [];
+
+  const publishEvent = {
+    type: ListItemEvent.PUBLISHED,
+    time: new Date('2022-01-01'),
+    jsonData: {
+      eventName: "publish",
+      userId: "smoke",
+    }
+  }
+
+  if(options.isPublished) {
+    events.push(publishEvent)
+  }
+
+  if(options.isArchived) {
+    events.push({
+      type: ListItemEvent.ARCHIVED,
+      jsonData: {
+        eventName: "archived",
+        userId: "smoke",
+      }
+    })
+  }
+
+  if(options.__smoke.isUnpublishedByUser === true) {
+    events.push(publishEvent, {
+      type: ListItemEvent.UNPUBLISHED,
+      jsonData: {
+        eventName: "unpublish",
+        userId: "smoke",
+      }
+    })
+  }
+
+  if(options.status === "ANNUAL_REVIEW_OVERDUE") {
+    events.push(publishEvent, {
+      type: ListItemEvent.UNPUBLISHED,
+      jsonData: {
+        eventName: "unpublish"
+      }
+    })
+  }
+
+  return events;
+}
+
 function listItem(options) {
-  const { jsonData, status, isPinned, service, ...rest } = options;
+  const { isArchived, emailVerified = true, isPublished, service = "lawyers", __smoke, ...rest } = options;
+
+  const events = setupPublishEvents(options);
+  const history = {
+    create: events
+  }
+
+  let status = options.status;
+
+  const jsonData = {
+    ...JSON_DATA[service](),
+    ...rest,
+    __smoke,
+    metadata: {
+      emailVerified
+    },
+  }
+
+  const updatedAt = options.updatedAt && new Date(options.updatedAt)
 
   return {
+    ...(updatedAt && {updatedAt}),
+    status,
+    isPublished,
     type: service,
-    jsonData: {
-      country: "Eurasia",
-      metadata: {
-        emailVerified: jsonData.metadata.emailVerified,
-      },
-      contactName: jsonData.contactName ?? randFullName(),
-      declaration: ["confirm"],
-      phoneNumber: "1234567",
-      emailAddress: jsonData.emailAddress ?? "ignoremyemail@noemail-ignoreme.uk",
-      publishEmail: "Yes",
-      speakEnglish: true,
-      websiteAddress: null,
-      organisationName: jsonData.organisationName ?? randCompanyName(),
-      representedBritishNationals: true,
-      ...jsonData,
-    },
+    jsonData,
     addressId: 329,
-    isApproved: false,
-    isPublished: rand([true, false]),
-    isBlocked: false,
-    status: status ?? rand(status),
-    ...rest,
+    ...(events.length && { history }),
   };
 }
 
-const status = [
+const STATUS = [
   "NEW",
   "OUT_WITH_PROVIDER",
   "EDITED",
   "ANNUAL_REVIEW",
-  "REVIEW_OVERDUE",
-  "REVIEWED",
+  "ANNUAL_REVIEW_OVERDUE",
   "PUBLISHED",
   "UNPUBLISHED",
 ];
```

Comment:
Should this include `CHECK_ANNUAL_REVIEW` as well?

Reply:
yep. This was originally intended to be used for randomised status, although I think all usages of this step end up defining a status anyway